modules/apps/journal/journal-service/src/main/java/com/liferay/journal/service/impl/JournalFolderLocalServiceImpl.java

```diff
@@ -1529,6 +1555,22 @@ private JournalFolder _getRestrictedAncestorFolder(JournalFolder folder)
 		return _getRestrictedAncestorFolder(folder.getParentFolder());
 	}
 
+	private void _validateExternalReferenceCode(
+			String externalReferenceCode, long groupId)
+		throws PortalException {
+
+		JournalFolder journalFolder = journalFolderPersistence.fetchByG_ERC(
+			groupId, externalReferenceCode);
+
+		if (journalFolder != null) {
+			throw new DuplicateFolderExternalReferenceCodeException(
```

Comment:
Just a note, it is not necessary to do it.
I would move this change and the deletion of the `DuplicateJournalFolderExternalReferenceCodeException` to the implementation and build service commits.

It is not necessary, you can put it here. Just a note for future PRs 🙂